content/en/docs/contributing/localization.md

```diff
@@ -204,22 +204,48 @@ For more details about the script, run `npm run check:i18n -- -h`.
 
 ## New localizations
 
-(Section To Be Completed soon with information about how to propose a new
-localization.)
+To start a new localization for the OpenTelemetry website,
+[raise an issue](https://github.com/open-telemetry/opentelemetry.io/issues/) to
+share your interest to contribute. Tag all other individuals, that are willing
+to write and review translations in the language you want to add. You need at
+least two potential contributors, three would be ideal. Include the following
+task list in your issue as well:
+
+- List at least 2 contributors for the new language: `<@gitHubHandle1>`,
+  `<@gitHubHandle2>`, ...
+- Localize index page to `<your language>`
+- Create an issue label for `lang:<shortcode>`
+- Create org-level group for `<shortcode>` approvers
+- Update components owners for `content/<shortcode>`
```

Comment:
How about having this as a Markdown code snippet they could paste in the issue?

Reply:
~~~suggestion
```
- [ ] List at least 2 contributors for the new language: `<@gitHubHandle1>`,
  `<@gitHubHandle2>`, ...
- [ ] Localize index page to `<your language>`
- [ ] Create an issue label for `lang:<shortcode>`
- [ ] Create org-level group for `<shortcode>` approvers
- [ ] Update components owners for `content/<shortcode>`
```
~~~

Reply:
done